openforcefield/tests/test_toolkits.py

```diff
@@ -2324,26 +2351,118 @@ def test_register_rdkit(self):
+        reason='RDKitToolkit or AmberToolsToolkit is not available')
+    def test_deregister_toolkit(self):
+        """Test removing an instantiated toolkit from the registry"""
+        toolkit_registry = ToolkitRegistry(toolkit_precedence=[AmberToolsToolkitWrapper, RDKitToolkitWrapper])
+
+        assert any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+        toolkit_registry.deregister_toolkit(toolkit_registry._toolkits[-1])
+        assert any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert not any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+        toolkit_registry.deregister_toolkit(toolkit_registry._toolkits[-1])
+        assert not any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert not any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+    @pytest.mark.skipif(
+        not RDKitToolkitWrapper.is_available() or not AmberToolsToolkitWrapper.is_available(),
+        reason='RDKitToolkit and AmberToolsToolkit not available')
+    def test_deregister_toolkit_by_class(self):
+        """Test removing a toolkit from the registry by matching class types"""
+        toolkit_registry = ToolkitRegistry(toolkit_precedence=[AmberToolsToolkitWrapper, RDKitToolkitWrapper])
+
+        assert any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+        toolkit_registry.deregister_toolkit(RDKitToolkitWrapper)
+        assert any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert not any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+        toolkit_registry.deregister_toolkit(AmberToolsToolkitWrapper)
+        assert not any([isinstance(tk, AmberToolsToolkitWrapper) for tk in toolkit_registry._toolkits])
+        assert not any([isinstance(tk, RDKitToolkitWrapper) for tk in toolkit_registry._toolkits])
+
+    @pytest.mark.skipif(
+        not RDKitToolkitWrapper.is_available() or not AmberToolsToolkitWrapper.is_available(),
+        reason='RDKitToolkit and AmberToolsToolkit not available')
+    def test_deregister_toolkit_bad_inputs(self):
+        """Test bad inputs to deregister_toolkit"""
+        toolkit_registry = ToolkitRegistry(toolkit_precedence=[AmberToolsToolkitWrapper])
+
+        with pytest.raises(InvalidToolkitError):
+            toolkit_registry.deregister_toolkit('rdkit as a string')
+
+        # Attempt to deregister a toolkit that is not registered
+        with pytest.raises(ToolkitUnavailableException):
+            toolkit_registry.deregister_toolkit(RDKitToolkitWrapper)
+
+    def deregister_from_global_registry(self):
```

Comment:
Yup. This is a pretty weird one. Hopefully this can be made more sensical when we move to a singleton design pattern or context manager approach.